opentelemetry-sdk/src/metrics/periodic_reader.rs

```diff
@@ -245,36 +245,56 @@ impl<RT: Runtime> PeriodicReaderWorker<RT> {
             Either::Left((res, _)) => {
                 res // return the status of export.
             }
-            Either::Right(_) => {
-                otel_error!(
-                    name: "collect_and_export",
-                    status = "timed_out"
-                );
-                Err(MetricsError::Other("export timed out".into()))
-            }
+            Either::Right(_) => Err(MetricsError::ExportTimeout(
+                "PeriodicReader".into(),
+                self.timeout.as_nanos(),
```

Comment:
Why are we converting this into nanoseconds?

Reply:
Also, we don't seem to be using the string or time duration in logging at all. Do we need them?

Reply:
we are doing it here

    #[error("Metrics reader {0} failed with timeout. Max configured timeout: {1} ns")]
    ExportTimeout(String, u128),

Reply:
Two questions here:
1. Why are we adding these new variants to the `MetricsError` enum? Would the end user be able to consume these variants? They only seem to be used in internal methods.
2. If we decide that these variants have to be public, why use nanoseconds for time out? Why not milliseconds?